jobserver/github.py

```diff
@@ -31,8 +20,58 @@ def get_file(repo, branch):
     ]
     f.args["ref"] = branch
 
-    headers = _build_headers(accept="application/vnd.github.3.raw")
+    headers = {
+        "Accept": "application/vnd.github.3.raw",
+        "Authorization": f"token {TOKEN}",
+        "User-Agent": USER_AGENT,
+    }
     r = requests.get(f.url, headers=headers)
     r.raise_for_status()
 
     return r.text
+
+
+def get_repos_with_branches():
+    """
+    Get Repos with their branches from the OpenSafely Researchers Team
+
+    This uses the GraphQL API to avoid making O(N) calls to GitHub's (v3) REST
+    API.
+    """
+    query = """
+    query {
+      organization(login: "opensafely") {
+        team(slug: "researchers") {
+          repositories(first: 100) {
```

Comment:
Isn't this `100` restriction going to get old very quickly?

Reply:
Yes, I think this gives us ~50 repos (IIRC 54) currently and I've yet to work out pagination with GraphQL.  But my hope was there would be enough breathing room to tackle the problem before it was an issue.

Thank you for the reminder though, I meant to make a ticket about it!

Reply:
Ah, GitHub says 58 [here](https://github.com/orgs/opensafely/teams/researchers/repositories).